playground/frontend/lib/modules/sdk/components/sdk_selector.dart

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import 'package:playground/config/theme.dart';
+import 'package:playground/modules/sdk/models/sdk.dart';
+
+typedef void SetSdk(SDK sdk);
+
+class SDKSelector extends StatelessWidget {
+  final SDK sdk;
+  final SetSdk setSdk;
+
+  const SDKSelector({Key? key, required this.sdk, required this.setSdk})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(
+        vertical: 0.0,
+        horizontal: 16.0,
+      ),
+      decoration: BoxDecoration(
+        color: ThemeColors.greyColor(context),
+        borderRadius: BorderRadius.circular(8.0),
+      ),
+      child: DropdownButtonHideUnderline(
+        child: DropdownButton<SDK>(
+          value: sdk,
+          icon: const Icon(Icons.keyboard_arrow_down),
+          iconSize: 24,
+          elevation: 1,
+          borderRadius: BorderRadius.circular(8.0),
+          alignment: Alignment.bottomCenter,
+          onChanged: (SDK? newSdk) {
+            if (newSdk != null) {
+              setSdk(newSdk);
```

Comment:
I think this could be improved by using a `Stateful` widget and using:

```dart
setState(() {
   this.sdk = ...
});
```

This would require redesigning the constructor of this class.  The `setState` is great if knowledge of the state is internal to the `Stateful` widget.  If knowledge of state is needed outside this `Stateful` widget, there are many options depending on whether these are sibling, child or widgets somewhere else in the Widget tree.  The easiest for parent/child relationships are using the https://api.flutter.dev/flutter/widgets/InheritedWidget-class.html.

I've been able to get away with more complex state sharing without using the popular [provider](https://pub.dev/packages/provider) package but I would have to find notes on this.

Reply:
Yes, we will need it outside this widget that's why I added the `provider` package and store SDK value at `PlaygroundState`.